### DISCUSSION TITLE
What's your development env?

### DISCUSSION THREAD

**Discussion by vitezprchal** [a-h/templ#885] [Q&A]
Hey, I'm finding my current development environment for go and go templ to be really slow.
I'm currently using air, gin, templ, tailwind and thats it.
Could you share what setup you're using that might help speed things up?

**Comment by nexovec** [comment]
Which part of it do you find slow?

**Comment by amoghyermalkar123** [comment]
@vitezprchal if you are referring to the slowness of the dev process itself, (like creating the project, building assets, reloading things) you can use a template like so - https://github.com/amoghyermalkar123/templ-quickstart
this is a fork ive added a makefile automation to.

**Comment by vitezprchal** [comment]
Specifically, when I recompile, my current stack using air & gin takes around ~2000ms just to refresh. Any suggestions would be appreciated

**Reply by joerdav** [reply]
Some additional information may help in this scenario please. If you have a repo that can be cloned we can see if the slow reload is due to the configuration or not. If this isn't possible then it could be useful for you to record a screen capture of the terminal as it reloads (with the `-v` flag enabled on any templ commands).

**Reply by gregmulvaney** [reply]
Go is a compiled language. Changes require recompiling the binary which isn't as quick as a hot reload on an interpreted language like JS

**Reply by vitezprchal** [reply]
Yes I'm aware it is compiled language I was thinking of making templ binding for javascript just for dev purposes

**Comment by jhelberg** [comment]
GNU emacs with LSP and tree-sitter works great. For building it's generate-lines and a makefile.

**Reply by cswank** [reply]
Do you happen to have an emacs configuration you could share?  I'm trying to learn how to use tree-sitter in emacs and I'm not sure where to start.

**Comment by nexovec** [comment]
I use helix for editing, and I also use air for hot reloading, like you all do. I develop in a container, it allows me to stage a network environment of the application and be a part of it, it's also a nice replicable environment. I'd much like to go with an embedded database like SQLite, because then I could run everything locally, but postgres is just so much more powerful.

A thing that can help with build time is to use
 
```toml
cmd = "go build -o ./executable -gcflags '-N -l'" # Unoptimized build
bin = "executable"
```
in your `.air.toml`.
